Fix #397. Add action to create Deployment Slots for Web App

sdubov opened the pull request:
Fixes #397

- Add an action to create a new Deployment Slot for a Web App from Azure Explorer node
<img width="365" alt="DeploymentSlotExplorer" src="https://user-images.githubusercontent.com/6064345/100326472-131dde80-2fdb-11eb-8bf2-1f06c1265894.png">

- Extend wrappers to list, create and check existence for a Deployment Slot inside an app
- Extend WebAppNode logic to collect information about Deployment Slot (if creating slots is support for an app). This should slow down a bit Web App node refresh because we need to grab information about web app pricing tier that does not exists on WebApp instance level
- Extend Web App and Function App validators logic to cover Deployment Slot support
- Add tests to cover new Deployemnt Slot logic
- Minor fixes in Logging

<img width="476" alt="DeploymentSlotDialog" src="https://user-images.githubusercontent.com/6064345/100326529-229d2780-2fdb-11eb-9c61-d27f3f7839e2.png">

sdubov committed:
Fix #397. Add action to create Deployment Slots for Web App

- Add an action to create a new Deployment Slot for a Web App from Azure Explorer node
- Extend wrappers to list, create and check existence for a Deployment Slot inside an app
- Extend WebAppNode logic to collect information about Deployment Slot (if creating slots is support for an app). This should slow down a bit Web App node refresh because we need to grab information about web app pricing tier that does not exists on WebApp instance level
- Extend Web App and Function App validators logic to cover Deployment Slot support
- Add tests to cover new Deployemnt Slot logic
- Minor fixes in Logging

maartenba reviewed (approved):
Have tested this, works nicely!

Two questions:
1) After creating a slot, the explorer node does not refresh automatically?
2) Is it expected that the first slot that I create has no option to "copy settings from"?
![image](https://user-images.githubusercontent.com/485230/100331245-32633e80-2fd0-11eb-9f59-d50e5d5a7543.png)

sdubov committed:
Add ability to close settings from production web app

- Add current web app to the list of clonable configurations for a Deployment Slot
- Fix refresh action after Deployment Slot is created

sdubov commented:
Thank you @maartenba. Fixed both issues.
1. Fixed node to refresh after a new deployment slot is created
2. Add current web app to the list of clonnable settings.
